src/shared/providers/auth.provider.tsx

```diff
@@ -1,19 +1,89 @@
-import React, { createContext, ReactNode } from 'react';
-import { createAuthStore, AuthState } from '@/shared/stores/auth.store';
+import React, {
+  createContext,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { useCloudStorage } from '@vkruglikov/react-telegram-web-app';
-import { UseBoundStore, StoreApi, Mutate } from 'zustand';
+import { User } from '@/shared/types/user.type';
+import { createUser } from '../api/auth.api';
 
-export const AuthContext = createContext<UseBoundStore<
-  Mutate<StoreApi<AuthState>, []>
-> | null>(null);
+export type AuthState = {
+  user?: User;
+  ready: boolean;
+  createUser: (user: Omit<User, 'id' | 'createdAt'>) => Promise<void>;
+  logoutUser: () => Promise<void>;
+};
+
+export const AuthContext = createContext<AuthState | undefined>(undefined);
 
 type AuthProviderProps = {
   children: ReactNode;
 };
 
 export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
   const { getItem, setItem } = useCloudStorage();
-  const store = createAuthStore(getItem, setItem);
 
-  return <AuthContext.Provider value={store}>{children}</AuthContext.Provider>;
+  const [store, setStore] = useState<AuthState>({
+    user: undefined,
+    ready: false,
+    createUser: async (user) => {
+      const newUser = await createUser(user);
+
+      if (newUser !== undefined) {
+        const newState = { ...store, user: newUser };
+        setStore(newState);
+      } else {
+        console.error('A problem ocurred when generating a user.');
+      }
+    },
+    logoutUser: async () => {
+      const newState = { ...store, user: undefined };
+      setStore(newState);
+    },
+  });
+
+  useEffect(() => {
+    setItem('auth', JSON.stringify(store));
+  }, [store]);
+
+  const isRehydrated = useRef(false);
+
+  useEffect(() => {
+    const rehydrate = async () => {
```

Comment:
Мне не оч нравится как эта хуета выглядит
```suggestion
    const rehydrate = async () => {
```